src/droplet.F

```diff
@@ -445,8 +468,73 @@ subroutine droplet_driver(dt,xh,uh,ruh,xf,yh,vh,rvh,yf,zh,mh,rmh,zf,mf,zs,    &
 
     !$acc update host(pdata,pdata_locind)
 
+    if (myid==0) write(*,*) 'TIME findloc:',t_findloc
+    if (myid==0) write(*,*) 'TIME interp:',t_interp
+
+    if (myid==0) write(*,*) 'TIME BEint:',t_BEint
+    if (myid==0) write(*,*) 'TIME BEprep:',t_BEprep
+    if (myid==0) write(*,*) 'TIME radsolver:',t_radsolver
+    if (myid==0) write(*,*) 'TIME gauss:',t_gauss
+    if (myid==0) write(*,*) 'TIME LV:',t_LV
+    if (myid==0) write(*,*) 'TIME neighbor:',t_neighbor
+
+    if (myid==0) write(*,*) 'TIME feed:',t_feed
+
+    call mpi_allreduce(mpi_in_place,num100,1,mpi_int,mpi_sum,mpi_comm_world,ierr)
+    call mpi_allreduce(mpi_in_place,num1000,1,mpi_int,mpi_sum,mpi_comm_world,ierr)
+    
+    if (myid==0) write(*,*) 'num100,num1000 = ',num100,num1000
+
     if(timestats.ge.1) time_droplet=time_droplet+mytime()
 
+    t_f = mpi_wtime()
+    if (myid==0) write(*,*) 'TIME nploop:',t_f-t_s
+    
+
+    t_s = mpi_wtime()
+
+!----------------------------------------------------------------------
+!  remove the "holes" due to droplets which fell out: collapse pdata and adjust nparcelsActive
+!  num_fallout already contains the number which have fallen out
+
+   allocate(holes_fallout_ind(num_fallout))
+
+   !List the spots in pdata which have "dead" particles
+   idx=1
+   do np=1,nparcelsActive
+
+      if (pdata(np,pract) .lt. 0.0) then  !If it has been flagged as falling out
+         holes_fallout_ind(idx) = np
+         idx=idx+1
+      end if
+
+   end do
+
+
+   !Now transfer these to the end of the pdata list
+   !Simply move the last particle in the list, pdata(nparcelsActive,:), to a hole and update nparcelsActive
+   do np=num_fallout,1,-1
+
+      if (holes_fallout_ind(np) .eq. nparcelsActive) then
+         nparcelsActive = nparcelsActive-1  !If the dead particle is at end, simply reduce nparcelsActive
+      else
+         pdata(holes_fallout_ind(np),:) = pdata(nparcelsActive,:)
+         pdata_locind(holes_fallout_ind(np),:) = pdata_locind(nparcelsActive,:)
+         pdata_neighbor(holes_fallout_ind(np)) = pdata_neighbor(nparcelsActive)
+         nparcelsActive = nparcelsActive-1
+      end if
+
+   end do
+
+   deallocate(holes_fallout_ind)
+
+   call mpi_allreduce(nparcelsActive,np_tmp,1,mpi_int,mpi_sum,mpi_comm_world,ierr)
```

Comment:
why do we need an `mpi_allreduce` call here?

Reply:
I have moved this down below. At the moment it's just for a writeout statement that lets me know how many particles are currently in the system (variable `np_tmp`). This can be removed later